Fix button for close the form And fix now can not create an empty user (minimun it needs the name) (PR from Fork Sergio's repos)

SergioAmadorLorente opened the pull request:
Fix event association and code in buttonClose for close properly the form in RentalHistoryForm.cs
&
Fix: Can create an empty user in EditCustomerForm.cs now it needs minimun insert a name to create it or to save it if its editing.

Added 2 labels to make it easy to the user to undestand it needs minimun a Name to create or edit.
![labelchanged](https://github.com/user-attachments/assets/9ff6f875-1d67-48f2-b6ee-fd1a7cca2b06)

SergioAmadorLorente committed:
Fix event association and code for close the form properly in RentalHistoryForm.cs
&
Fix can create an empty user in EditCustomerForm.cs now it needs minimun insert a name to create it

nach8D-edu commented on HobbyManiaManager/Forms/Rental/RentalHistoryForm.cs:
```diff
@@ -43,7 +43,8 @@ private RentalDataGridViewModel BuildRentalViewModel(Rental r)
         private void buttonClose_Click(object sender, EventArgs e)
         {
-            Dispose();
+            this.Close();
```
I don't understand the reason behind this change.

SergioAmadorLorente replied:
Because if dowload the main and you press than button will not work. Because in the event section of the buttonClose 
 event Click its empty, so I linked it with buttonClose_Click and I use this.Close() because its more familiar for me. But I can change to Dispose(); will work too now because the event link its correct now.

nach8D-edu commented on HobbyManiaManager/Forms/Customer/EditCustomerForm.cs:
```diff
@@ -147,6 +147,14 @@ private void buttonCancel_Click(object sender, EventArgs e)
+            else 
+            {
+                buttonUpdateCreate.Enabled = false;
+            }
```
I would suggest something cleaner like:
`buttonUpdateCreate.Enabled = !string.IsNullOrWhiteSpace(textBoxName.Text);
`
Instead of the if.

SergioAmadorLorente replied:
I am testing your code and works fine. Its cleaner than my code, sorry for my noob code still learning  heheheh I will do PR soon

nach8D-edu reviewed (changes requested):
I added some comments. Please check them. In addition, I don't understand why there are changes in *.Design.cs files.

SergioAmadorLorente committed:
Commit fixing with clean code and adding label to make easy for the user undestand its mandatory field the name for  create or edit a user.

SergioAmadorLorente commented:
I added new PR;
- Changed the code for the clean code.
- I added a label in EditCustomerForm.cs so that the user can now easily understands that the name is mandatory (for Create and Update too), check it if you are agree or not.
- I know github said I changed the Designs in my first PR but I didin't probably github error. I only changed EditCustomerForm.cs (adding the label probably now will change de desing) and RentalHistoryForm.cs thanks

nach8D-edu reviewed (changes requested):
The change in the *.Designer.cs files are because of the labels you created and the event link.

Please add a screenshooyt in the PR description in order to evaluate the visual changes (labels) you added.

SergioAmadorLorente commented:
![labelchanged](https://github.com/user-attachments/assets/6801c41c-fa06-4e53-b8d8-19eb11a83d3e)
